gcp/app-engine/slack_handler.py

```diff
@@ -0,0 +1,271 @@
+                status_code = 400
+        else:
+            player_search = words[-2]  # Assume second last word is player name/id
+            tournament_search = words[-1]  # Assume the last word is the tournament name/id
+
+        if not response_text:
+            player_info = get_player(player_search, tournament_search, round)
+            if player_info:
+                try:
+                    # Create and save the image
+                    text = f"*{player_info['name']}* in *{player_info['tournament_name']}* (Round {player_info['round']}):"
+                    image = create_golf_scorecard_image(player_info)
+                    response_text, status_code = save_and_upload_slack_image(image, channel, text=text)
+                except Exception as e:
+                    response_text = f'Error creating image: {e}'
+                    status_code = 500
+            else:
+                response_text = "Player or tournament not found."
+                status_code = 404
+    else:
+        response_text = "Please specify both a player and a tournament."
+        status_code = 400
+
+    if response_text:
+        slack_client.chat_postMessage(channel=channel, text=response_text)
+    return response_text, status_code
+
+def handle_top_10_players(command_text, channel):
+    words = command_text.split()
+    status_code = 200
+    if len(words) > 3:
+        tournament_search = words[-1]  # Assume the last word is the tournament name/id
+        top_10_players = get_top_players(tournament_search, limit=10)
+        if top_10_players:
+            response_text = "*Top 10 Players:*\n"
+            for player in top_10_players:
+                response_text += (
+                    f"{player['rank']}. {player['name']} - Score: {player['score']}\n"
+                )
+        else:
+            response_text = "Tournament not found."
+            status_code = 404
+    else:
+        response_text = "Please specify a tournament."
+        status_code = 400
+
+    slack_client.chat_postMessage(
+        channel=channel,
+        text=response_text
+    )
+    return response_text, status_code
+
+def format_scores_grid(pars, scores):
+    grid = "```\n"
+    grid += " Hole  |" + "|".join([" " + str(i).ljust(3) for i in range(1, 19)]) + "|\n"
+    grid += "-------|" + "|".join(["----" for _ in range(1, 19)]) + "|\n"
+    grid += " Par   |" + "|".join([" " + str(par).ljust(3) for par in pars]) + "|\n"
+    grid += "-------|" + "|".join(["----" for _ in range(1, 19)]) + "|\n"
+    grid += " Score |" + "|".join([" " + str(score).ljust(3) for score in scores]) + "|\n```"
+    return grid
+
+# "player_info" is of the format:
+# {
+#   "name": "player name",
+#   "tournament_name": "tournament name",
+#   "round": 1-4,
+#   "pars": [parHole1, parHole2,...],
+#   "scores": [scoreHole1, scoreHole2,...]
+# }
+def create_golf_scorecard_image(player_info):
```

Comment:
fancy method that chat gpt helped me create to generate images for a scorecard, e.g. 
![ac5394e8-b13a-44e7-ae9d-87e913541677](https://github.com/user-attachments/assets/fd356293-c8e3-45dd-ad47-3cf5b41f9633)

Reply:
ok i have updated this to support better than birdie and worse than bogey, e.g.
![image](https://github.com/user-attachments/assets/a78948f3-a64a-499f-be6f-d345bab11ba5)